ts/src/main/scala/ch/epfl/ts/engine/MarketRules.scala

```diff
@@ -111,14 +117,25 @@ class MarketRules extends Serializable {
         }
 
         // Update price
-        bestMatch.price
+        result = bestMatch.price
 
         // no match found
       } else {
         println("MS: no match found - enqueuing")
         enqueueOrElse(newOrder, newOrdersBook)
-        oldTradingPrice
+        result = oldTradingPrice
       }
     }
+    generateQuote(marketId, newOrdersBook, bestMatchsBook, newOrder.timestamp, send)
+    result
+  }
+  def generateQuote(marketId: Long, bids: PartialOrderBook, asks: PartialOrderBook, timestamp: Long, send: Streamable => Unit) = {
+    if (!bids.isEmpty && !asks.isEmpty){
+      val topBid = bids.head
+      val topAsk = asks.head
+      println("MR: generating quote " + topBid.price + " " + topAsk.price)
+      send(Quote(marketId, timestamp, topAsk.whatC, topAsk.withC, topBid.price, topAsk.price))
+    } else
+      println("MR: can't generate quote")
```

Comment:
1. this is the expected behavior
2. it can be (at least I don't see how to do it) - see the comment about changing to `log.info` above.